### PR TITLE
SDK changes to support QIR submitters

### PR DESCRIPTION
**Pull request by cesarzc**
Changes needed to the SDK to support QIR submitters.

### CODE REVIEW

File: src/QuantumSdk/Sdk/Sdk.props [hunk 1]
```diff
@@ -27,13 +27,13 @@
...
+    <PackageReference Condition="$(ResolvedProcessorArchitecture.Contains('QCI')) And $(IncludeProviderPackages)" Include="Microsoft.Quantum.Providers.QCI" Version="0.24.201332" />
     <!-- Target packages included for specific execution targets. -->
-    <PackageReference Condition="'$(ResolvedQuantumIntrinsics)' == 'Type1' And $(ResolvedProcessorArchitecture.Contains('QirProcessor'))" Include="Microsoft.Quantum.Type1.Core" Version="0.24.201332" IsTargetPackage='true' />
-    <PackageReference Condition="'$(ResolvedQuantumIntrinsics)' == 'Type2' And $(ResolvedProcessorArchitecture.Contains('QirProcessor'))" Include="Microsoft.Quantum.Type2.Core" Version="0.24.201332" IsTargetPackage='true' />
-    <PackageReference Condition="'$(ResolvedQuantumIntrinsics)' == 'Type3' And $(ResolvedProcessorArchitecture.Contains('QirProcessor'))" Include="Microsoft.Quantum.Type3.Core" Version="0.24.201332" IsTargetPackage='true' />
+    <PackageReference Condition="'$(ResolvedQuantumIntrinsics)' == 'Type1' And $(ResolvedProcessorArchitecture.Contains('QirProcessor'))" Include="Microsoft.Quantum.Type1.Core" Version="0.24.201332" IsTargetPackage="true" ExcludeAssets="Compile" />
```
Comment (swernli) [inline src/QuantumSdk/Sdk/Sdk.props:34]: Can you remind me again why we ended up needing to add the `ExcludeAssets="Compile"` to the target package inclusion? Was that to keep the C# compilation happy?

Comment (cesarzc) [reply src/QuantumSdk/Sdk/Sdk.props:34]: Yes, it was to keep C# compilation happy :)

**Review by swernli — APPROVED** [review]

File: src/QuantumSdk/DefaultItems/DefaultItems.targets [hunk 2]
```diff
@@ -44,17 +44,18 @@
...
     -->
-    <ResolvedProcessorArchitecture Condition="('$(ResolvedQSharpOutputType)' == 'QSharpExe' Or '$(NoEntryPoint)' == 'true') And $(ExecutionTarget.ToLowerInvariant().StartsWith('honeywell.qir'))">HoneywellQirProcessor</ResolvedProcessorArchitecture>
-    <ResolvedProcessorArchitecture Condition="('$(ResolvedQSharpOutputType)' == 'QSharpExe' Or '$(NoEntryPoint)' == 'true') And $(ExecutionTarget.ToLowerInvariant().StartsWith('qci.qir'))">QCIQirProcessor</ResolvedProcessorArchitecture>
+    <!-- TODO: This might be temporary while it is decided how QIR targets are going to be handled. -->
+    <ResolvedProcessorArchitecture Condition="('$(ResolvedQSharpOutputType)' == 'QSharpExe' Or '$(NoEntryPoint)' == 'true') And $(ExecutionTarget.ToLowerInvariant().StartsWith('quantinuum'))">QuantinuumQirProcessor</ResolvedProcessorArchitecture>
+    <ResolvedProcessorArchitecture Condition="('$(ResolvedQSharpOutputType)' == 'QSharpExe' Or '$(NoEntryPoint)' == 'true') And $(ExecutionTarget.ToLowerInvariant().StartsWith('qci'))">QCIQirProcessor</ResolvedProcessorArchitecture>
```
Comment (swernli) [inline src/QuantumSdk/DefaultItems/DefaultItems.targets:49]: Actually, before merging, we need to make sure the mechanism for opting into QIR is well understood. As written, these lines will override for any Quantinuum or QCI targets and force usage of the target packages below, which may not work with the existing service translation logic.

**Review by swernli — CHANGES_REQUESTED** [review]
We need to ensure that the `QirProcessor` flavored targets are opt-in instead of overriding the existing logic.

File: src/QuantumSdk/DefaultItems/DefaultItems.targets (hunk 2 above)
Comment (cesarzc) [reply src/QuantumSdk/DefaultItems/DefaultItems.targets:49]: Yes, we do need to do that before merging into `main` but this is currently targeting a feature branch.
@bettinaheim, you mentioned that you were working on something that also needed SDK changes that was going to be merged into a feature branch. Please let me know if you want this PR to target a different branch.

Comment (bettinaheim) [reply src/QuantumSdk/DefaultItems/DefaultItems.targets:49]: Let's merge this into the feature branch, and I'll update the Sdk handling to have both coexist peacefully as part of also propagating the necessary information into the compiler.

Comment (swernli) [reply src/QuantumSdk/DefaultItems/DefaultItems.targets:49]: ah, good point, I didn't notice it was to a feature branch!

**Review by swernli — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

**Comment by bettinaheim** [comment]
@cesarzc I am closing this PR since I believe it is no longer relevant/needed.

**Pull request closed without merging**